bmorph/evaluation/simple_river_network.py

```diff
@@ -610,3 +615,53 @@ def generate_weight_map(self):
             area_fraction = area/total_area
             weight_map.append(f'{i}-{area_fraction}')
         return weight_map
+    
+#In Progress (below):    
+    def color_network_graph(self,measure,cmap):
+        if not measure.empty:
+            return plotting.color_code_nxgraph(self.network_graph,measure,cmap);
+        else:
+            color_bar = None;
+            segs = np.arange(0,len(self.seg_id_values));
+            color_vals = segs/len(segs);
+            color_dict =  {f'{seg}': mpl.colors.to_hex(cmap(i)) for i, seg in zip(color_vals, segs)};
+            return color_dict, color_bar
+        
+    def size_network_graph(self,measure):
+        segs = np.arange(0,len(self.seg_id_values));
+        size_vals = segs/len(segs);
+        size_dict = {f'{seg}': 200*size_vals(i) for i, seg in zip(size_vals,segs)};
+        return size_dict   
+        
+    def draw_network(self,label_map=[], color_measure=None, cmap = mpl.cm.get_cmap('hsv'), 
+                     node_size = 200, font_size = 8, font_weight = 'bold', node_shape = 's', linewidths = 2, font_color = 'w', node_color = None,
+                     with_labels=False,with_cbar=False,with_background=True):
+        
+        network_color_dict, network_color_cbar = self.color_network_graph(color_measure,cmap)
+        
+        if len(label_map) > 0:
+            new_network_color_dict = dict()
+            for key in network_color_dict.keys():
+                new_network_color_dict[f"{label_map[int(key)]}"] = network_color_dict[key]
+
+            new_network_graph = nx.relabel_nodes(self.network_graph,dict(zip(self.network_graph.nodes(),label_map)),copy=True)
+
+            network_color_dict = new_network_color_dict
+            self.network_graph = new_network_graph
+            self.network_positions = plotting.organize_nxgraph(self.network_graph)
+                    
+        network_nodecolors = [network_color_dict[f'{node}'] for node in self.network_graph.nodes()]
+        if node_color:
+            network_nodecolors = node_color;
+            
+        #network_node_dict = self.size_network_graph(None);
+        
```

Comment:
Are these lines needed/instructional at all? If yes, could you provide a comment. If no, could you remove?